### ISSUE TITLE
python setup.py build error redefinition of ‘PyMPI_MPICH_MPI_Type_get_extent_x

### ISSUE THREAD

**Issue by Xingyu-Romantic** [mpi4py/mpi4py#537]
gcc (GCC) 7.5.0
GNU Fortran (GCC) 4.8.5 20150623
mpiexec --version  Version:3.0

```bash
building 'mpi4py.MPI' extension
creating build/temp.linux-x86_64-cpython-39
creating build/temp.linux-x86_64-cpython-39/src
creating build/temp.linux-x86_64-cpython-39/src/mpi4py
/data/dnn/luxy/mpich/bin/mpicc -pthread -B /data/share_ana/compiler_compat -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /data/share_ana/include -I/data/share_ana/include -fPIC -O2 -isystem /data/share_ana/include -fPIC -DHAVE_DLFCN_H=1 -DHAVE_DLOPEN=1 -Isrc -I/data/share_ana/include/python3.9 -c src/mpi4py/MPI.c -o build/temp.linux-x86_64-cpython-39/src/mpi4py/MPI.o
In file included from src/lib-mpi/compat.h:10:0,
                 from src/mpi4py/MPI.c:1253:
src/lib-mpi/compat/mpich3.h:116:12: error: redefinition of ‘PyMPI_MPICH_MPI_Type_get_extent_x’
 static int PyMPI_MPICH_MPI_Type_get_extent_x(MPI_Datatype datatype,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/lib-mpi/compat/mpich3.h:10:12: note: previous definition of ‘PyMPI_MPICH_MPI_Type_get_extent_x’ was here
 static int PyMPI_MPICH_MPI_Type_get_extent_x(MPI_Datatype datatype,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/lib-mpi/compat/mpich3.h:160:12: warning: ‘PyMPI_MPICH_MPI_Rget_accumulate’ defined but not used [-Wunused-function]
 static int PyMPI_MPICH_MPI_Rget_accumulate(const void *origin_addr,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/lib-mpi/compat/mpich3.h:136:12: warning: ‘PyMPI_MPICH_MPI_Get_accumulate’ defined but not used [-Wunused-function]
 static int PyMPI_MPICH_MPI_Get_accumulate(const void *origin_addr,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/lib-mpi/compat/mpich3.h:10:12: warning: ‘PyMPI_MPICH_MPI_Type_get_extent_x’ defined but not used [-Wunused-function]
 static int PyMPI_MPICH_MPI_Type_get_extent_x(MPI_Datatype datatype,
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: command '/data/dnn/luxy/mpich/bin/mpicc' failed with exit code 1

```

**Comment by dalcinl** [comment]
Indeed, this is an mpi4py bug trying to workaround issues for MPICH version 3.0 that was released 12 years ago.

**Comment by dalcinl** [comment]
Tentative fix in branch [`bugfix/mpich-3.0`](https://github.com/mpi4py/mpi4py/tree/bugfix/mpich-3.0)

@Xingyu-Romantic Could you please try my changes?  For your convenience, the quickest way would be
```console
pip install --no-cache-dir https://github.com/mpi4py/mpi4py/archive/bugfix/mpich-3.0.tar.gz
```
```

**Comment by Xingyu-Romantic** [comment]
> Tentative fix in branch [`bugfix/mpich-3.0`](https://github.com/mpi4py/mpi4py/tree/bugfix/mpich-3.0)
> 
> @Xingyu-Romantic Could you please try my changes? For your convenience, the quickest way would be
> 
> ```
> pip install --no-cache-dir https://github.com/mpi4py/mpi4py/archive/bugfix/mpich-3.0.tar.gz
> ```

thanks for your replay, i will try it in next Monday

**Comment by dalcinl** [comment]
@Xingyu-Romantic Any update from your side?
